pkg/lifecycle/once.go

```diff
@@ -108,33 +98,35 @@ func Once() LifecycleOnce {
 // Start will run the `f` function once and return the error.
 // If Start is called multiple times it will return the error
 // from the first time it was called.
-func (l *lifecycleOnce) Start(f func() error) error {
-	if l.state.CAS(int32(Idle), int32(Starting)) {
+func (o *Once) Start(f func() error) error {
+	if o.state.CAS(int32(Idle), int32(Starting)) {
 		var err error
 		if f != nil {
 			err = f()
 		}
 
 		// skip forward to error state
 		if err != nil {
-			l.setError(err)
-			l.state.Store(int32(Errored))
-			close(l.stoppingCh)
-			close(l.stopCh)
+			o.setError(err)
+			o.state.Store(int32(Errored))
+			close(o.stoppingCh)
+			close(o.stopCh)
 		} else {
-			l.state.Store(int32(Running))
+			o.state.Store(int32(Running))
 		}
-		close(l.startCh)
+		close(o.startCh)
 
 		return err
 	}
 
-	<-l.startCh
-	return l.loadError()
+	<-o.startCh
+	return o.loadError()
 }
 
-func (l *lifecycleOnce) WhenRunning(ctx context.Context) error {
-	state := LifecycleState(l.state.Load())
+// WhenRunning blocks until the instance enters the running state, or the
+// context times out.
+func (o *Once) WhenRunning(ctx context.Context) error {
```

Comment:
Can we cite a precedent or anti-precedent?

- WaitGroup.Wait() is a precedent for Wait*.
- time.Until does not wait.
- Running is better than Start, since we are waiting for the Running state, not the Start() call (implied Started state).
- There is no precedent for a When method in Golang.
- I favored WhenRunning for brevity. I dislike having to memorize modifiers in method names, like "For", which could just as well be "Until".